pkg/controllers/hpareplicassyncer/hpa_replicas_syncer_controller.go

```diff
@@ -0,0 +1,148 @@
+package hpareplicassyncer
+
+import (
+	"context"
+
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	autoscalingv2 "k8s.io/api/autoscaling/v2"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/scale"
+	"k8s.io/klog/v2"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+var HPAPredicate = predicate.Funcs{
+	CreateFunc: func(e event.CreateEvent) bool {
+		return false
+	},
+	UpdateFunc: func(e event.UpdateEvent) bool {
+		oldHPA, ok := e.ObjectOld.(*autoscalingv2.HorizontalPodAutoscaler)
+		if !ok {
+			return false
+		}
+
+		newHPA, ok := e.ObjectNew.(*autoscalingv2.HorizontalPodAutoscaler)
+		if !ok {
+			return false
+		}
+
+		return oldHPA.Status.CurrentReplicas != newHPA.Status.CurrentReplicas
+	},
+	DeleteFunc: func(e event.DeleteEvent) bool {
+		return true
```

Comment:
Now do we not need to care about the deletion event?

Reply:
Done.